### PR TITLE
Summarize a rule and its contents in an HTML

### PR DESCRIPTION
**Pull request by matejak**
#### Description:

The new utility is able to display compiled rule fields next to each other.

#### Rationale:

This can be used to make sure that the rule is consistent.

To test it, you need to compile a product first.
You will also need the `pygments` Python package, you can optionally generate a highlight stylesheet (as advised in the help message).
Consider using this with data introduced by #9574 - policy-specific fields are displayed in the appropriate position.

### CODE REVIEW

File: utils/render-rule.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        built_content_path = pathlib.Path(self.built_content_path)
+        bash_loc = built_content_path / "fixes" / "bash" / (rule_id + ".sh")
+        fixes.bash = self._highlight_file(bash_loc, pygments.lexers.BashLexer())
+        ansible_loc = built_content_path / "fixes" / "ansible" / (rule_id + ".yml")
+        fixes.ansible = self._highlight_file(ansible_loc, pygments.lexers.YamlLexer())
+
+        checks = Namespace()
+        oval_loc = built_content_path / "checks" / "oval" / (rule_id + ".xml")
+        checks.oval = self._highlight_file(oval_loc, pygments.lexers.XmlLexer())
+
```
Comment (jan-cerny) [inline utils/render-rule.py:74]: If a rule doesn't have the check or doesn't have a remediation, it fails and emits a traceback.

File: utils/render-rule.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        fixes = Namespace()
+        built_content_path = pathlib.Path(self.built_content_path)
+        bash_loc = built_content_path / "fixes" / "bash" / (rule_id + ".sh")
+        fixes.bash = self._highlight_file(bash_loc, pygments.lexers.BashLexer())
+        ansible_loc = built_content_path / "fixes" / "ansible" / (rule_id + ".yml")
+        fixes.ansible = self._highlight_file(ansible_loc, pygments.lexers.YamlLexer())
+
+        checks = Namespace()
+        oval_loc = built_content_path / "checks" / "oval" / (rule_id + ".xml")
```
Comment (jan-cerny) [inline utils/render-rule.py:72]: Also keep in mind that some rules are templated so they use a different directory.

File: utils/rendering/rule-template.html [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	{{% endif %}}
+
+	{{%- if fixes.contains_something() %}}
+	<div class="checks">
+	<h2>Fixes</h2>
+	{{{ minor_section("Bash", fixes.bash, collapsible=true) -}}}
+
+	{{{ minor_section("Ansible", fixes.ansible, collapsible=true) -}}}
+
+	{{{ render_policy_specific_content(fixes) -}}}
```
Comment (jan-cerny) [inline utils/rendering/rule-template.html:106]: How do I get this working?

**Review by jan-cerny — CHANGES_REQUESTED** [review]

File: utils/rendering/rule-template.html (hunk 3 above)
Comment (matejak) [reply utils/rendering/rule-template.html:106]: This will work with data from #9574

File: utils/render-rule.py (hunk 1 above)
Comment (matejak) [reply utils/render-rule.py]: Not any more

File: utils/render-rule.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            sys.exit(1)
+
+        prose = Namespace()
+        prose.description = common.resolve_var_substitutions(rule.description)
+        prose.rationale = common.resolve_var_substitutions(rule.rationale)
+
+        fixes = Namespace()
+        built_content_path = pathlib.Path(self.built_content_path)
+        bash_loc = built_content_path / "fixes" / "bash" / (rule_id + ".sh")
+        fixes.bash = self._highlight_file(bash_loc, pygments.lexers.BashLexer())
```
[the remark this answers is not in the document]
Comment (matejak) [reply utils/render-rule.py]: It works differently in remediations, where it is not breaking stuff s.a. `sub` elements in the prose. So I find it OK-ish even without processing.

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**